pkg/kubefed2/federate/util.go

```diff
@@ -18,15 +18,33 @@ package federate
 
 import (
 	"fmt"
+	"io"
+	"os"
 	"strings"
 
 	apiextv1b1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/rest"
 )
 
+func DecodeYAMLFromFile(filename string, obj interface{}) error {
+	f, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	return DecodeYAML(f, obj)
```

Comment:
In case of error, does it need to be supplemented with a wrapper with message more relevant. I am not sure what strings `decode()` returns.

Reply:
I think the caller should be responsible for wrapping the error.  I've updated the call in `enable.go` accordingly.  The only other usage is in tests and I think that can evolve as needed.